pkg/build/admission/defaults/admission.go

```diff
@@ -71,6 +71,28 @@ func (b BuildDefaults) applyPodDefaults(pod *kapi.Pod) {
 	for k, v := range b.config.Annotations {
 		addDefaultAnnotations(k, v, pod.Annotations)
 	}
+
+	// Apply default resources
+	defaultResources := b.config.Resources
+	if len(pod.Spec.Containers[0].Resources.Limits) == 0 {
```

Comment:
Resources.Limits is a ResourceList which is a map[ResourceName]resource.Quantity, from what i've seen the length of a nil map is zero:
https://play.golang.org/p/oiK99llO_y

As for containers[0], yes for now build pods only have exactly one container always.  This will change when we move to using init containers but for now it's safe.